client/core/core.go

```diff
@@ -1899,6 +1899,78 @@ func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg
 		return err
 	}
 
+	// If there are active trades, make sure they can be settled by the
+	// keys held within the new wallet.
+	sameWallet := func() error {
+		ownsAddr := func(addr string) error {
+			owns, err := wallet.OwnsAddress(addr)
+			if err != nil {
+				return err
+			}
+			if !owns {
+				return fmt.Errorf("new wallet does not own address found in active trades: %v", addr)
+			}
+			return nil
+		}
+		for _, dc := range c.dexConnections() {
+			maybeDifferentWallet := false
+			for _, trade := range dc.trackedTrades() {
+				if !trade.isActive() {
+					continue
+				}
+				waID := wallet.AssetID
+				// If the to asset, check if we own an
+				// contract.Address.
+				if trade.wallets.toAsset.ID == waID {
+					if err := ownsAddr(trade.Trade().SwapAddress()); err != nil {
+						return err
+					}
+					// Assume all trade addresses are
+					// owned by the new wallet if one is.
+					return nil
+				}
+				// If the from asset, check if we own a
+				// refund address for a match if any exist.
+				if trade.wallets.fromAsset.ID == waID {
+					for _, match := range trade.matches {
+						script := match.MetaData.Proof.Script
+						if len(script) == 0 {
+							continue
+						}
+						addr, err := wallet.RefundAddress(script)
+						if err != nil {
+							return err
+						}
+						if err := ownsAddr(addr); err != nil {
+							return err
+						}
+						// Assume all refund addresses
+						// are owned by the new
+						// wallet if one is.
+						return nil
+					}
+					// If we did not find a refund address,
+					// we cannot be sure that this is the
+					// same wallet.
+					//
+					// TODO: Implement a way to check that
+					// these accounts are the same or not.
```

Comment:
Epic indent here! :smile: I think I might have made a package level function or dexConnection method taking an xcWallet instead of a closure, but this is fine.